File: frontend/src/features/checkins/types/checkIn.ts
export interface CreateCheckInRequest {
  trailId: string;
  completedDate: string;
  notes?: string;
}

export interface UpdateCheckInRequest {
  completedDate: string;
  notes?: string;
}

export interface CheckInResponse {
  id: string;
  userId: string;
  trailId: string;
  completedDate: string;
  notes?: string | null;
  isHidden: boolean;
}

File: backend/Entities/CheckIn.cs
namespace backend.Entities;

public class CheckIn
{
    public Guid Id { get; set; }

    public Guid UserId { get; set; }

    public Guid TrailId { get; set; }

    public DateTime CompletedDate { get; set; }

    public string? Notes { get; set; }

    public string? PhotoUrl { get; set; }

    public bool IsHidden { get; set; }

    public User User { get; set; } = null!;

    public Trail Trail { get; set; } = null!;
}
